src/components/Modal.ts

```diff
@@ -0,0 +1,133 @@
+	            <label for="distance text-caption">거리(도보 이동 시간) </label>
+	            <select name="distance" id="distance" required>
+	              <option value="">선택해 주세요</option>
+	              <option value="5">5분 내</option>
+	              <option value="10">10분 내</option>
+	              <option value="15">15분 내</option>
+	              <option value="20">20분 내</option>
+	              <option value="30">30분 내</option>
+	            </select>
+	          </div>
+	
+	          <!-- 설명 -->
+	          <div class="form-item">
+	            <label for="description text-caption">설명</label>
+	            <textarea name="description" id="description" cols="30" rows="5"></textarea>
+	            <span class="help-text text-caption">메뉴 등 추가 정보를 입력해 주세요.</span>
+	          </div>
+	
+	          <!-- 링크 -->
+	          <div class="form-item">
+	            <label for="link text-caption">참고 링크</label>
+	            <input type="text" name="link" id="link">
+	            <span class="help-text text-caption">매장 정보를 확인할 수 있는 링크를 입력해 주세요.</span>
+	          </div>
+	
+	          <!-- 취소/추가 버튼 -->
+	          <div class="button-container">
+	            <button type="button" class="button button--secondary text-caption">취소하기</button>
+	            <button class="button button--primary text-caption">추가하기</button>
+	          </div>
+	        </form>
+	      </div>
+	    </div>
+	    `;
+  }
+
+  render(): void {
+    this.$target.insertAdjacentHTML('beforeend', this.template());
+    this.listenEvent();
+  }
+
+  listenEvent() {
+    this.$target.querySelector('.button--primary')?.addEventListener('click', (event: Event) => {
+      event.preventDefault();
+
+      const category: Category = this.$target.querySelector<HTMLSelectElement>('#category')!.value as Category;
+      const name: string = this.$target.querySelector<HTMLInputElement>('#name')!.value;
+      const distance: Distance = Number(this.$target.querySelector<HTMLSelectElement>('#distance')!.value) as Distance;
+      const description = this.$target.querySelector<HTMLTextAreaElement>('#description')!.value;
+      const link = this.$target.querySelector<HTMLInputElement>('#link')!.value;
+
+      if (!category || !name || !distance) {
+        alert('카테고리, 이름, 거리는 필수 입력 정보 입니다!');
+        return;
+      }
+
+      const restaurant: Restaurant = {
+        name,
+        category,
+        distance,
+        description,
+        link,
+      };
+
+      this.restaurants.add(restaurant);
+      this.$state.restaurants = this.restaurants.getRestaurants();
+
+      localStorage.setItem('state', JSON.stringify(this.$state));
+
+      this.closeModal();
+    });
+
+    this.$target.querySelector('.button--secondary')?.addEventListener('click', () => {
+      this.closeModal();
+    });
+  }
+
+  closeModal() {
+    const elementToRemove = this.$target.querySelector('.modal--open');
+
+    if (elementToRemove) {
+      elementToRemove.remove();
+    }
```

Comment:
모달창을 닫을 때 아예 제거하는군요.  
CSS 상태를 조절하는 방법도 고려해보셨을까요?

Reply:
막연하게 전부 없애버리는게 좋지 않을까 생각했었습니다😅 
css를 조절하는 방법으로 수정했습니다
```suggestion
    if (modal) {
      modal.classList.add('hidden');
    }
```